src/controls/qml/HighlightBar.qml

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2023 - Timo Könnecke <github.com/eLtMosen>
+ * Copyright (C) 2022 - Ed Beroset <github.com/beroset>
+ * Copyright (C) 2020 - Darrel Griët <idanlcontact@gmail.com>
+ * Copyright (C) 2015 - Florent Revest <revestflo@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.9
+import org.asteroid.controls 1.0
+
+Rectangle {
+    id: highlightBar
+
+    anchors.fill: parent
+    color: rowClick.containsPress ? "#33ffffff" : "#00ffffff"
+
+    Behavior on color {
+        ColorAnimation {
+            duration: 150;
+            easing.type: Easing.InCubic
+        }
+    }
+
+    MouseArea {
+        id: rowClick
+
+        property bool check: checked
+        anchors.fill: parent
+        hoverEnabled: true
+        onClicked: {
+            toggle.checked = !toggle.checked
```

Comment:
What is toggle here ? If that's a reference to an id that is supposed to exist outside of HighlightBar then this shouldn't be implemented like this since this breaks the rules of encapsulation. Rather the onClicked property should be exposed (with an alias for example) and the HightlightBar users should reimplement this function referring to their toggle.

Reply:
Yeah, i failed implementing that. If i get onClicked exposed, i can also use the Highlightbar in ListItem and LabeledActionbutton. Which was the initial plan for making a HighlightBar component at all.
Continue trying. But the error i faced is that the the alias was read-only for some reason.

Reply:
This has been solved thanks to @beroset help. The onClick binding needed a function.
Now the HighlightBar is correctly receiving onClicks from the parents.
LabeledSwitch, LabeledActionButton and ListView now all use this highlightBar component